backend/api/Controllers/MissionController.cs

```diff
@@ -141,20 +216,114 @@ public async Task<ActionResult<byte[]>> GetMap([FromRoute] string assetCode, str
     }
 
     /// <summary>
-    /// Schedule a new mission
+    /// Reschedule an existing mission
     /// </summary>
     /// <remarks>
-    /// <para> This query schedules a new mission and adds it to the database </para>
+    /// <para> This query reschedules an existing mission and adds it to the database </para>
+    /// </remarks>
+    [HttpPost("{missionId}/reschedule")]
+    [Authorize(Roles = Role.User)]
+    [ProducesResponseType(typeof(MissionRun), StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<MissionRun>> Rechedule(
+        [FromRoute] string missionId,
+        [FromBody] RescheduleMissionQuery rescheduledMissionQuery
+    )
+    {
+        var robot = await _robotService.ReadById(rescheduledMissionQuery.RobotId);
+        if (robot is null)
+            return NotFound($"Could not find robot with id {rescheduledMissionQuery.RobotId}");
+
+        MissionDefinition? missionDefinition;
+        try
+        {
+            missionDefinition = await _missionDefinitionService.ReadById(missionId);
+            if (missionDefinition == null)
+                return NotFound("Mission definition not found");
+        }
+        catch (HttpRequestException e)
+        {
+            if (e.StatusCode.HasValue && (int)e.StatusCode.Value == 404)
+            {
+                _logger.LogWarning(
+                    "Could not find mission definition with id={id}",
+                    missionId
+                );
+                return NotFound("Mission definition not found");
+            }
+
+            _logger.LogError(e, "Error getting mission database");
+            return StatusCode(StatusCodes.Status502BadGateway, $"{e.Message}");
+        }
+
+        List<MissionTask>? missionTasks;
+        switch (missionDefinition.Source.Type)
+        {
+            case MissionSourceType.Echo:
+                missionTasks = _echoService.GetMissionByPath(missionDefinition.Source.URL).Result.Tags
+                    .Select(
+                        t =>
+                        {
+                            var tagPosition = _stidService
+                                .GetTagPosition(t.TagId, missionDefinition.AssetCode)
+                                .Result;
+                            return new MissionTask(t, tagPosition);
+                        }
+                    )
+                    .ToList();
+                break;
+            case MissionSourceType.Custom:
+                missionTasks = _sourceService.GetMissionTasksFromURL(missionDefinition.Source.URL);
+                break;
+            default:
+                return BadRequest("Invalid mission source type provided");
```

Comment:
Do an internal server error here. This bad request cannot happen as the MissionSourceType is collected from the mission definition itself and not from the request.